addons/test_mail/tests/test_mail_activity.py

```diff
@@ -29,6 +29,79 @@ def setUpClass(cls):
         ])
 
 
+@tests.tagged('mail_activity')
+class TestActivitySearch(TestActivityCommon):
+
+    def test_search_on_date_done_includes_archived(self):
+        """
+        Test that searching on 'date_done' on mail.activity includes archived (done)
+        activities, as implemented in the custom _search method.
+        """
+        active_activity = self.env['mail.activity'].create({
+            'res_model_id': self.env['ir.model']._get_id('mail.test.activity'),
+            'res_id': self.test_record.id,
+            'user_id': self.user_employee.id,
+            'activity_type_id': self.env.ref('mail.mail_activity_data_todo').id,
+            'summary': 'Active To-Do',
+            'date_deadline': '2025-09-01',
+        })
+
+        done_activity = self.env['mail.activity'].create({
+            'res_model_id': self.env['ir.model']._get_id('mail.test.activity'),
+            'res_id': self.test_record.id,
+            'user_id': self.user_employee.id,
+            'activity_type_id': self.env.ref('mail.mail_activity_data_todo').id,
+            'summary': 'Done To-Do',
+            'date_deadline': '2025-09-02',
+        })
+        done_activity.action_done()
+
+        archived_record = self.env['mail.test.activity'].create({'name': 'Archived Test Record', 'active': False})
+        done_activity_on_archived_record = self.env['mail.activity'].create({
+            'res_model_id': self.env['ir.model']._get_id('mail.test.activity'),
+            'res_id': archived_record.id,
+            'user_id': self.user_employee.id,
+            'activity_type_id': self.env.ref('mail.mail_activity_data_todo').id,
+            'summary': 'Done To-Do on Archived Record',
+            'date_deadline': '2025-09-03',
+        })
+        done_activity_on_archived_record.action_done()
+
+        self.assertEqual(self.env['mail.activity'].search([]), active_activity)
```

Comment:
You could have a sort of `base_domain = ['id', 'in', activities.ids]`, and then just append to that in the rest of the test. This way there's no chance a lingering mail.activity will be matched (either because you're running the test with data while developing, or because of data from setUpClass, ...)